### PR TITLE
fix: populate qr code correctly

### LINKED ISSUE #2739: QR scanner not populating address in Send Modal
### Overview

When trying to scan a QR code for a send on mobile, or on web, in release.shapeshift.com the address does not populate in the send modal. However, if you select next, the transaction will build with the address from the QR code.
Release: https://github.com/shapeshift/web/pull/2733

### References and additional details

1) log into release.shapeshift.com on mobile or web and connect to wallet of choice
2) Select an asset and hit the send button
3) Select the QR scanner option and allow the site to access camera
4) Scan the QR code of an acceptable wallet address
5) Notice the address does not show in the send modal
6) Select Next and notice the transaction build with the address of the QR code

### Acceptance Criteria

Address should populate in send modal after QR code has been scanned

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Ownership

- [X] If my bounty needs engineering or needs product I have added the respective labels on the right
- [X] As the sponsor of this bounty I will review the changes in a preview environment (ops/product) or review the PR (engineering)

### Estimated effort

_No response_

### Sponsor / Stakeholder

_No response_

### Bounty Hunters

- [Join our discord](https://discord.gg/shapeshift)
- Include an expected timeline for you to complete work in the work plan when you apply for this bounty!
- Please refer to [this link](https://shapeshiftdao.gitbook.io/getting-started/how-to-contribute/bounties#resources-for-new-contributors-and-bounty-hunters) for some basic info
- Please ***do not*** start work on this issue until you are approved in Gitcoin.

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
## Description

i broke this on the send receive multi account branch playing whack a mole fixing a dangerous bug where switching an asset would not change the to address, which could potentially be on the wrong chain.

the `useEffect` to clear the input field on asset change was not the correct way to do this.

we now clear the `Input` field when we handle the asset selection.

this causes the validation handler on the input field to run and clear the hidden/internal `Address` and
`VanityAddress` fields to be reset.

additionally, this moves the trim logic from the input `onChange` handler into the validation function such that all the biz logic is in one place, avoid a react error about changing an input element from an uncontrolled to controlled
component.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #2739

## Risk

* this could have broken sending for any or all assets

## Testing

* open the send modal
* scan a qr code
* go all the way through to the confirm screen - ensure the `Send To` address is populated correctly
* go back to the send details page (where you type amounts)
* change the asset to something different
* it will take you back to the address input page which should be cleared with no console errors, and the `Next` button should be disabled

## Screenshots (if applicable)

n/a - no visual changes, qr scanner should work again

### CODE REVIEW

File: src/components/Modals/Send/AddressInput/AddressInput.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ export const AddressInput = ({ rules }: AddressInputProps) => {
             spellCheck={false}
             autoFocus
             fontSize='sm'
-            onChange={e => onChange(e.target.value.trim())}
+            onChange={e => onChange(e)}
```
Comment (0xdef1cafe) [inline src/components/Modals/Send/AddressInput/AddressInput.tsx:38]: moved into validation handler

File: src/components/Modals/Send/Form.tsx [hunk 2]
```diff
@@ -62,6 +62,7 @@ export const Form: React.FC<SendFormProps> = ({ asset: initialAsset, accountId }
 
   const handleAssetSelect = async (asset: Asset) => {
     methods.setValue(SendFormFields.Asset, { ...asset, ...marketData })
+    methods.setValue(SendFormFields.Input, '')
```
Comment (0xdef1cafe) [inline src/components/Modals/Send/Form.tsx:65]: reset the user facing `Input` field on asset change, which in tun resets the internal `Address` and `VanityAddress` fields used later

File: src/components/Modals/Send/views/Address.tsx [hunk 3]
```diff
@@ -37,13 +37,11 @@ export const Address = () => {
     formState: { errors },
   } = useFormContext<SendInput>()
   const address = useWatch<SendInput, SendFormFields.Address>({ name: SendFormFields.Address })
+  const input = useWatch<SendInput, SendFormFields.Input>({ name: SendFormFields.Input })
   const { send } = useModal()
   const asset = useWatch<SendInput, SendFormFields.Asset>({ name: SendFormFields.Asset })
   const isYatFeatureEnabled = useFeatureFlag('Yat')
 
-  // reset address when asset changes
```
Comment (0xdef1cafe) [inline src/components/Modals/Send/views/Address.tsx:44]: this was correctly clearing the field when the asset changed, but also causing the scanned QR code to erroneously clear

File: src/plugins/cosmos/components/modals/Send/SendCommon.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 export enum SendFormFields {
+  Input = 'input', // the raw field input on the address input
```
Comment (0xdef1cafe) [inline src/plugins/cosmos/components/modals/Send/SendCommon.ts:2]: this was required to be added to the cosmos version of the send form to satisfy the compiler, as the `Address` component is shared between them

File: src/components/Modals/Send/AddressInput/AddressInput.tsx (hunk 1 above)
Comment (0xApotheosis) [inline src/components/Modals/Send/AddressInput/AddressInput.tsx:38]: We can tighten this.

```suggestion
            onChange={onChange}
```

**Review by 0xApotheosis — APPROVED** [review]
- Tested that we still can build send transactions from both a manually entered address, and entered via a QR code
- Changing the asset after scanning a QR code clears the receive address and requires re-entry before proceeding with the flow

LGTM.

**Review by 0xApotheosis — APPROVED** [review]

**Pull request merged**